### PR TITLE
[@types/jquery.validation] Adding customElements to ValidationOptions

### PR DESCRIPTION
**Pull request by dhobi**
jquery.validation 1.21.0 (https://github.com/jquery-validation/jquery-validation/releases/tag/1.21.0) has added support for registering additional elements to be validatable (for example web components)

This PR adds this customElements to ValidationOptions

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jquery-validation/jquery-validation/pull/2493
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

File: types/jquery.validation/index.d.ts
Comment (avidenic) [inline types/jquery.validation/index.d.ts:26]: Interestingly this is present in the source code, but not in the documentation.

**Review by avidenic — APPROVED** [review]

File: types/jquery.validation/index.d.ts
Comment (dhobi) [reply types/jquery.validation/index.d.ts:26]: Yes, true. I just opened a PR to fix this in the documentation as well https://github.com/jquery-validation/validation-content/pull/59. I'll wait for an approval there and will merge this PR once the documentation PR is merged. Thanks for your review! ❤️

**Comment by avidenic** [comment]
Ready to merge

**Pull request merged**
